plugin/core/types.py

```diff
@@ -275,11 +276,14 @@ def r(name: str, default: Union[bool, int, str, list, dict]) -> None:
             r("inhibit_snippet_completions", False)
             r("inhibit_word_completions", True)
 
-        # Backwards-compatible with "diagnostics_highlight_style"
-        diagnostics_highlight_style = s.get("diagnostics_highlight_style")
-        if isinstance(diagnostics_highlight_style, str):
-            if not diagnostics_highlight_style:
-                self.show_diagnostics_highlights = False
+        # correctness checking will happen inside diagnostics_highlight_style_flags method
+        self.diagnostics_highlight_style = s.get("diagnostics_highlight_style")
+
+        # Backwards-compatible with "show_diagnostics_highlights"
+        if self.diagnostics_highlight_style is None:
+            show_diagnostics_highlights = s.get("show_diagnostics_highlights")
+            self.diagnostics_highlight_style = "default" if show_diagnostics_highlights else ""
```

Comment:
Not sure if it makes sense to have `"default"` here, it is not a special value. The alternative is to specify the full dict, but that would mean duplicating logic between this and `DIAGNOSTIC_SEVERITY` inside `views.py`.

Reply:
I think the only thing you need to do here is to set all styles to `""` if "show_diagnostics_highlights" is `False`. Otherwise the default values are already defined by the default setting for "diagnostics_highlight_style".

Reply:
Oh right, the settings file we ship already has the proper defaults.